VideoAPI/Testing.Common/Helper/ZAP.cs

```diff
@@ -0,0 +1,168 @@
+                } 
+            }
+
+            throw new Exception(string.Format("Waited for {0} minute(s), however could not access the API successfully, hence could not verify if ZAP started successfully or not", minutesToWait));
+        }
+
+        public static void Scan(string url)
+        {
+            if (!ZapConfiguration.RunZap) return;
+
+            if (ZapConfiguration.RunSpider)
+            {
+                StartSpidering(url);
+            }
+
+            if (ZapConfiguration.RunActiveScan)
+            {
+                StartActiveScanning(url);
+            }
+        }
+
+        private static void StartSpidering(string target)
+        {
+            _apiResponse = _api.spider.scan(target, "", "", "", "");
+            var scanid = ((ApiResponseElement)_apiResponse).Value;
+            PollTheSpiderTillCompletion(scanid);
+        }
+
+        private static void PollTheSpiderTillCompletion(string scanid)
+        {
+            int spiderProgress;
+            while (true)
+            {
+                Sleep(1000);
+                spiderProgress = int.Parse(((ApiResponseElement)_api.spider.status(scanid)).Value);
+                if (spiderProgress >= 100)
+                    break;
+            }
+
+            Console.WriteLine("Spider complete");
+            Sleep(10000);
+        }
+        private static void StartActiveScanning(string target)
+        {
+            _apiResponse = _api.ascan.scan(target, "", "", "", "", "", "");
+
+            string activeScanId = ((ApiResponseElement)_apiResponse).Value;
+            PollTheActiveScannerTillCompletion(activeScanId);
+        }
+        private static void PollTheActiveScannerTillCompletion(string activeScanId)
+        {
+            int activeScannerprogress;
+            while (true)
+            {
+                Sleep(5000);
+                activeScannerprogress = int.Parse(((ApiResponseElement)_api.ascan.status(activeScanId)).Value);
+                if (activeScannerprogress >= 100)
+                    break;
+            }
+        }
+
+
+        private static void Sleep(int milliseconds)
+        {
+            do
+            {
+                Thread.Sleep(milliseconds);
+                milliseconds = milliseconds - 2000;
+            } while (milliseconds > 2000);
+        }
+
+        public static void ReportAndShutDown(string reportFileName)
+        {
+            if (!ZapConfiguration.RunZap) return;
+
+            WriteHtmlReport(reportFileName);
+            WriteXmlReport(reportFileName);
+            ShutdownZAP();
+        }
+
+        private static void WriteHtmlReport(string reportFileName)
+        {
+            File.WriteAllBytes(reportFileName + ".html", _api.core.htmlreport());
+        }
+
+        private static void WriteXmlReport(string reportFileName)
+        {
+            File.WriteAllBytes(reportFileName + ".xml", _api.core.xmlreport());
+        }
+
+        private static void ShutdownZAP()
```

Comment:
Maybe implement an Idisposable class to handle shutdown properly

Reply:
Static call and now wrapped it under Finally which is called on OneTimeTearDown so should be safely disposed.